pkg/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -0,0 +1,288 @@
+package machineconfig
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/coreos/go-systemd/unit"
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	performancev1alpha1 "github.com/openshift-kni/performance-addon-operators/pkg/apis/performance/v1alpha1"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	defaultIgnitionVersion       = "2.2.0"
+	defaultFileSystem            = "root"
+	defaultIgnitionContentSource = "data:text/plain;charset=utf-8;base64"
+)
+
+const (
+	rtKernel       = "rt-kernel"
+	preBootTuning  = "pre-boot-tuning"
+	reboot         = "reboot"
+	bashScriptsDir = "/usr/local/bin"
+)
+
+const (
+	systemdSectionUnit     = "Unit"
+	systemdSectionService  = "Service"
+	systemdSectionInstall  = "Install"
+	systemdDescription     = "Description"
+	systemdWants           = "Wants"
+	systemdAfter           = "After"
+	systemdBefore          = "Before"
+	systemdEnvironment     = "Environment"
+	systemdType            = "Type"
+	systemdRemainAfterExit = "RemainAfterExit"
+	systemdExecStart       = "ExecStart"
+	systemdWantedBy        = "WantedBy"
+)
+
+const (
+	systemdServiceKubelet      = "kubelet.service"
+	systemdServiceTypeOneshot  = "oneshot"
+	systemdTargetMultiUser     = "multi-user.target"
+	systemdTargetNetworkOnline = "network-online.target"
+	systemdTrue                = "true"
+)
+
+const (
+	environmentRTRepoURL       = "RT_REPO_URL"
+	environmentNonIsolatedCpus = "NON_ISOLATED_CPUS"
+)
+
+// NewPerformance returns new machine configuration object for performance sensetive workflows
+func NewPerformance(assetsDir string, profile *performancev1alpha1.PerformanceProfile) (*machineconfigv1.MachineConfig, error) {
+	name := components.GetComponentName(profile.Name, components.RoleWorkerPerformance)
+	mc := &machineconfigv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: machineconfigv1.GroupVersion.String(),
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				components.LabelMachineConfigurationRole: name,
+			},
+		},
+		Spec: machineconfigv1.MachineConfigSpec{},
+	}
+
+	ignitionConfig, err := getIgnitionConfig(assetsDir, *profile.Spec.RealTimeKernel.RepoURL, profile.Spec.CPU.NonIsolated)
+	if err != nil {
+		return nil, err
+	}
+
+	mc.Spec.Config = *ignitionConfig
+	mc.Spec.KernelArguments = getKernelArgs(profile.Spec.HugePages, profile.Spec.CPU.Isolated)
+
+	return mc, nil
+}
+
+func getKernelArgs(hugePages *performancev1alpha1.HugePages, isolatedCPUs *performancev1alpha1.CPUSet) []string {
+	kargs := []string{
```

Comment:
Can we pass these values in the CR as well ?  Can we and do we want to pass then in the CR ?

Reply:
Currently, I pass the whole performance profile.